CLI/src/main/java/main/Drifty_CLI.java

```diff
@@ -61,6 +62,51 @@ public static void main(String[] args) {
                         help();
                         System.exit(0);
                     }
+                    case ADD_FLAG -> {
+                        if (i + 1 < args.length && Utility.isURL(args[i + 1])) {
+                            boolean isUrlValid;
+                            if (Utility.isURL(args[i + 1])) {
+                                isUrlValid = Utility.isLinkValid(args[i + 1]);
+                            } else {
+                                isUrlValid = false;
+                                messageBroker.msgLinkError(INVALID_LINK);
+                            }
+                            if (!isUrlValid) {
+                                LOGGER.log(MessageType.INFO, CLI_APPLICATION_TERMINATED);
+                                System.exit(0);
+                            }
+                            addUrlToFile(args[i + 1]);
+                            System.out.println("URL added: " + args[i + 1]);
+                            listUrls();
+                        } else {
+                            messageBroker.msgInitError("No URL provided or invalid URL format.");
+                        }
+                        System.exit(0);
+                    }
+                    case LIST_FLAG -> {
+                        listUrls();
+                        System.exit(0);
+                    }
+                    case REMOVE_FLAG -> {
+                        if (i + 1 < args.length) {
+                            try {
+                                int index = Integer.parseInt(args[i + 1]);
+                                removeUrl(index);
```

Comment:
As you implement the YAML format, you need to change the way `removeURL()` method works.